node/src/components/linear_chain.rs

```diff
@@ -185,49 +250,43 @@ where
                         },
                     )
             }
-            Event::IsBonded(Some(mut signatures), fs, true) => {
-                // Known block and signature from a bonded validator.
-                // Check if we had already seen this signature before.
-                let signature_known = signatures
-                    .proofs
-                    .get(&fs.public_key)
-                    .iter()
-                    .any(|sig| *sig == &fs.signature);
-                // If new, gossip and store.
-                if signature_known {
-                    self.remove_from_pending_fs(&*fs);
-                    Effects::new()
-                } else {
-                    let mut effects = effect_builder
-                        .announce_finality_signature(fs.clone())
-                        .ignore();
-                    if let Some(signature) = self.remove_from_pending_fs(&*fs) {
-                        if signature.is_local() {
-                            let message = Message::FinalitySignature(fs.clone());
-                            effects.extend(effect_builder.broadcast_message(message).ignore());
-                        }
-                    };
-                    signatures.insert_proof(fs.public_key, fs.signature);
-                    // Cache the results in case we receive the same finality signature before we
-                    // manage to store it in the database.
-                    self.cache_signatures(*signatures.clone());
-                    debug!(hash=%signatures.block_hash, "storing finality signatures");
-                    effects.extend(
-                        effect_builder
-                            .put_signatures_to_storage(*signatures)
-                            .ignore(),
-                    );
-                    effects
-                }
+            Event::IsBonded(Some(mut known_signatures), fs, true) => {
+                // New finality signature from a bonded validator.
+                known_signatures.insert_proof(fs.public_key, fs.signature);
+                // Cache the results in case we receive the same finality signature before we
+                // manage to store it in the database.
+                self.linear_chain_state
+                    .cache_signatures(*known_signatures.clone());
+                debug!(hash=%known_signatures.block_hash, "storing finality signatures");
```

Comment:
```suggestion
                debug!(hash = %known_signatures.block_hash, "storing finality signatures");
```

Reply:
Done in [`5ffb8ab` (#1268)](https://github.com/CasperLabs/casper-node/pull/1268/commits/5ffb8ab4bee2a6e7972a9678ef4a3b7ccfc57dd3)